### ISSUE TITLE
Consider whether to build Electron apps for macOS

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js-windows#441]
TLDR: Probably not worth it at this point in time, but an option for the future?

Electron-builder (and electron-forge) allow building apps for a range of platforms, including macOS (but not iOS). GitHub Actions provide a VM on which the app can be built easily, though signing and notarizing is a challenge and might well need to be set up and tested on a local Mac first (which I don't have) before the configuration is set up on GitHub Actions.

There are pros and cons.

Pros:

* I already build nightly for Windows and Linux; adding macOS (DMG or PKG) is just an extra step and it would be completely automated via CI.
* It would increase the reach of the Electron app and provide alternatives (especially for the packaged apps WikiMed and Wikivoyage) for Mac users.
* Choice is good!

Cons:

* I probably can't set it up without access to a Mac.
* There is already a macOS Kiwix app developed by a heroic solitary developer. I don't want to undermine or detract from his work. Any port should be seen as complementary and adding value to the ecosystem, not fragmenting it.
* Electron doesn't support iOS.
* There is arguably no need, as the PWA can be installed on macOS and iOS already (although clearly this doesn't allow for packaged apps or offline installation).
* It would use up developer time that could be better deployed on more urgent issues.

**Comment by kelson42** [comment]
@Jaifroid We will have an effort to push custom apps on iOS in the next months. Considering this is the same codebase, I wonder how this push could apply to macOS too?

**Comment by Jaifroid** [comment]
If you're pushing for custom apps on iOS, then I presume you'd get custom apps on macOS "for free", as it is the same codebase over there for both OS's, as you say. The Electron solution would only be complementary if in fact it's a different process to make custom apps for iOS and for macOS. Kiwix JS Electron could only help with the macOS side, as Electron does not run on iOS.

Anyway, I don't want to tread on anyone's toes. Maybe keep this possibility in reserve in case it's needed in the future.

**Comment by lacymorrow** [comment]
You can build for mac using GH actions, I'd also like a mac version as it isn't immediately obvious from the website that there is a portable mac version. There is, here https://download.kiwix.org/release/kiwix-desktop-macos/

**Comment by Jaifroid** [comment]
Thanks. The issue is I don't have a Mac to test on.... However, as you've discovered, there is a Kiwix Desktop Mac version, so unless there are specific features of the Electron version that are needed on Mac, and that can't be supplied simply by using and installing the PWA in Chrome, it maybe seems a bit unnecessary to build a specific Mac version?

**Comment by lacymorrow** [comment]
Yep, likely. Would allow discontinuing the Mac native version to make maintenance easier. I'm only using it for navigating offline copies of wikipedia so I don't think the PWA version will work on a USB stick

**Comment by Jaifroid** [comment]
@lacymorrow No, the PWA won't work on a USB stick on a system on which it has never been installed. However, once installed in the OS's browser (Safari, Chrome), or bookmarked (Firefox), it will be able to work offline and will be able to pick archives from the USB stick. The native Desktop macOS app should be fine, though I'm not sure if that needs to be installed as well. At least it will definitely support offline installation.

**Comment by lacymorrow** [comment]
Makes sense, I appreciate the response. The intent is to load USB sticks with a dump of wikipedia and an offline browser to send to folks who don't have internet, so the PWA version won't work there

**Comment by Jaifroid** [comment]
OK, so I suppose the kiwix-desktop macOS version is fine for that usage, and then either the Electron or desktop versions of the Windows and Linux apps. To build and (above all, sign, which isn't an easy process) a macOS version of the Electron app, I think I'd need a Mac-specialist collaborator on this Repo...
